pkg/kube/inject/inject_test.go

```diff
@@ -692,3 +693,63 @@ func TestAppendMultusNetwork(t *testing.T) {
 		})
 	}
 }
+
+func TestParseInjectEnvs(t *testing.T) {
+	cases := []struct {
+		name string
+		in   string
+		want map[string]string
+	}{
+		{
+			name: "empty",
+			in:   "/",
+			want: map[string]string{},
+		},
+		{
+			name: "no-kv",
+			in:   "/inject",
+			want: map[string]string{},
+		},
+		{
+			name: "no-kv-with-tail",
+			in:   "/inject/",
+			want: map[string]string{},
+		},
+		{
+			name: "one-kv",
+			in:   "/inject/cluster/cluster1",
+			want: map[string]string{"ISTIO_META_CLUSTER_ID": "cluster1"},
+		},
+		{
+			name: "two-kv",
+			in:   "/inject/cluster/cluster1/net/network1/",
+			want: map[string]string{"ISTIO_META_CLUSTER_ID": "cluster1", "ISTIO_META_NETWORK": "network1"},
+		},
+		{
+			name: "not-predefined-kv",
+			in:   "/inject/cluster/cluster1/network/network1",
+			want: map[string]string{"ISTIO_META_CLUSTER_ID": "cluster1"},
+		},
+		{
+			name: "one-key-without-value",
+			in:   "/inject/cluster",
+			want: map[string]string{},
+		},
+		{
+			name: "key-without-value",
+			in:   "/inject/cluster/cluster1/network",
+			want: map[string]string{"ISTIO_META_CLUSTER_ID": "cluster1"},
+		},
+	}
+
+	for i := range cases {
+		tc := cases[i]
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
nit: since this is a super fast test parallel may just make things more complicated. I generally only use it if the test is doing slow IO

Reply:
sure, done